upstream_utils/upstream_utils.py

```diff
@@ -372,6 +372,8 @@ def reset(self):
 
         subprocess.run(["git", "switch", "--detach", self.old_tag])
 
+        self.set_root_tag(self.old_tag)
```

Comment:
Is there a reason this is before apply_patches() instead of after like in rebase()?

Reply:
No- The reason I wrote it that way was that I based it off of `clone()` (with the idea that `reset()` is like `clone()` plus adding the patches) instead of off of `rebase(self.old_tag)` (which is how it was originally described).
Speaking of, was there a particular reason to set the tag after `apply_patches()` in `rebase()`?

Reply:
I think it can be either way. It just needs to be consistent.